src/calendar.rs

```
@@ -0,0 +1,171 @@
+
+        let rule = AclRule {
+            role: Some("reader".into()),
+            scope: Some(AclRuleScope {
+                type_: Some("default".into()),
+                ..Default::default()
+            }),
+            ..Default::default()
+        };
+        if let Err(why) = self
+            .calendar_hub
+            .acl()
+            .insert(rule, calendar.id.as_ref().unwrap_or_else(|| unreachable!()))
+            .doit()
+            .await
+        {
+            error!(?why, "Failed to set acl for the calendar");
+        };
+        calendar
+    }
+
+    #[instrument(skip(self))]
+    pub async fn delete_calendar(&self, calendar_id: &str) {
+        if let Err(why) = self
+            .calendar_hub
+            .calendars()
+            .delete(calendar_id)
+            .doit()
+            .await
+        {
+            error!(?why, "Failed to delete the calendar");
+        };
+    }
+
+    #[instrument(skip(self))]
+    pub async fn create_event(&self, event: Event, calendar_id: &str) {
+        if let Err(why) = self
+            .calendar_hub
+            .events()
+            .insert(event, calendar_id)
+            .doit()
+            .await
+        {
+            error!(?why, "Failed to create an event")
+        };
+    }
+
+    #[instrument(skip(self))]
+    pub async fn delete_event(&self, id: &str, calendar_id: &str) {
+        if let Err(why) = self
+            .calendar_hub
+            .events()
+            .delete(calendar_id, id)
+            .doit()
+            .await
+        {
+            error!(?why, "Failed to delete an event")
+        };
+    }
+
+    #[instrument(skip(self))]
+    pub async fn list_calendars(&self) -> Vec<CalendarListEntry> {
+        match self.calendar_hub.calendar_list().list().doit().await {
+            Ok(response) => response.1.items.unwrap_or_else(|| unreachable!()),
+            Err(why) => {
+                error!(?why, "Failed to list calendars");
+                vec![]
```

return error instead of default value